src/test/java/com/kodilla/ecommercee/repositoryDao/CartRepositoryTestSuit.java

```diff
@@ -0,0 +1,200 @@
+        Assert.assertNotEquals(null,cartRepository.findById(cart1.getId()));
+        Assert.assertNotEquals(null,cartRepository.findById(cart2.getId()));
+        Assert.assertNotEquals(null,cartRepository.findById(cart3.getId()));
+        //Clean Up
+        try{
+            cartRepository.deleteById(cart1.getId());
+            cartRepository.deleteById(cart2.getId());
+            cartRepository.deleteById(cart3.getId());
+        }catch (Exception e){
+            LOGGER.error("Clean-up process failed.", e.getMessage(), e);
+        }
+    }
+    @Transactional
+    @Test
+    public void testCartDaoSaveAndFindById(){
+        //Given
+        cartRepository.save(cart1);
+        cartRepository.save(cart2);
+        cartRepository.save(cart3);
+        Long IdCart1 = cart1.getId();
+        //When
+        Optional<Cart> cartFound = cartRepository.findById(IdCart1);
+        //Then
+        Assert.assertEquals(IdCart1,cartFound.get().getId());
+        //Clean Up
+        try{
+            cartRepository.deleteById(cart1.getId());
+            cartRepository.deleteById(cart2.getId());
+            cartRepository.deleteById(cart3.getId());
+        }catch (Exception e){
+            LOGGER.error("Clean-up process failed.", e.getMessage(), e);
+        }
+    }
+    @Transactional
+    @Test
+    public void testCartDaoDeleteById(){
+        //Given
+        cartRepository.save(cart1);
+        cartRepository.save(cart2);
+        cartRepository.save(cart3);
+        Long cart2Id = cart2.getId();
+        //When
+        cartRepository.deleteById(cart2Id);
+        List<Cart> cartList = cartRepository.findAll();
+        //Then
+        Assert.assertFalse(cartList.contains(cart2));
+        //Clean Up
+        try{
+            cartRepository.deleteById(cart1.getId());
+            cartRepository.deleteById(cart3.getId());
+        }catch (Exception e){
+            LOGGER.error("Clean-up process failed.", e.getMessage(), e);
+        }
+    }
+
+    @Transactional
+    @Test
+    public void testCartOneOnOneWithOrder(){
+        //Given
+        Order order = new Order();
+        cart1.setOrder(order);
+        //When
+        cartRepository.save(cart1);
+        Assert.assertNotEquals(zeroLong,cartRepository.findById(cart1.getOrder().getOrderId()));
+        //Clean Up
+        try{
+            cartRepository.deleteById(cart1.getId());
+        }catch (Exception e){
+            LOGGER.error("Clean-up process failed.", e.getMessage(), e);
+        }
+    }
+    @Transactional
+    @Test
+    public void testCartManyToOneUser(){
+        //Given
+        user1 = new Users();
+        user1.getCarts().add(cart1);
+        user1.getCarts().add(cart2);
+        cart1.setUsers(user1);
+        cart2.setUsers(user1);
+        //When
+        userDao.save(user1);
+        cartRepository.save(cart1);
+        cartRepository.save(cart2);
+        //Then
+        Assert.assertEquals(2,user1.getCarts().size());
```

Comment:
Przed sprawdzeniem pobierz Usera z bazy - teraz sprawdzasz obiekt ktory sam zbudowales w Given

Reply:
@erwinMuz przeciez zrobilem to tak samo jak Ola w testach produkt:
  ```
@Test
    public void testProductDaoManyToOneGroup(){
        //GIVEN
        Group group = new Group("ubrania");
        group.getProductList().add(product1);
        group.getProductList().add(product2);
        product1.setGroup(group);
        product2.setGroup(group);
        //WHEN
        groupRepository.save(group);
        productRepository.save(product1);
        productRepository.save(product2);
        //THEN
        Assert.assertEquals(2, group.getProductList().size());

przeciez tutaj tez porownuje obiekt ktory sama zbudowala a mimo wszystko kazdy zatwierdzil i zostal dolaczony do mastera

Reply:
Sluszna uwaga, test Oli tez jest do poprawy.